### PR TITLE
Refactor version calculations for CMake and CPack - omit initial v, etc.

### LINKED ISSUE #637: v0.8.0-alpha2 .deb packages are not installable -- version does not begin with a number
As reported on Gitter.

### PR DESCRIPTION
**Pull request by stephengtuggy**
Thank you for submitting a pull request and becoming a contributor to the Vega Strike Core Engine.

Please answer the following:

Code Changes:
- [ ] Have the PR Validation Tests been run? See https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Pull-Request-Validation
- [ ] This is a documentation change only
- [X] This is an installer/build system change only

Issues:
- Please list any related issues
- Fixes #637 

Purpose:
- What is this pull request trying to do? Fix the version numbers that did not conform to Debian's version numbering scheme
- What release is this for? 0.8.x
- Is there a project or milestone we should apply this to? 0.8.x

### CODE REVIEW

File: engine/CMakeLists.txt [hunk 1]
```diff
@@ -33,24 +34,33 @@
...
+            COMMAND git rev-parse --short HEAD
+            WORKING_DIRECTORY ${GIT_ROOT_DIR}
+            OUTPUT_VARIABLE VEGASTRIKE_VERSION_TWEAK
+            OUTPUT_STRIP_TRAILING_WHITESPACE
     )
+    SET(VEGASTRIKE_VERSION_TWEAK "-$VEGASTRIKE_VERSION_TWEAK")
 ENDIF ()
-
+SET(VEGASTRIKE_VERSION_LONG "${VEGASTRIKE_VERSION_MAJOR}.${VEGASTRIKE_VERSION_MINOR}.${VEGASTRIKE_VERSION_PATCH}${VEGASTRIKE_VERSION_TWEAK}")
+SET(VEGASTRIKE_VERSION_SHORT "${VEGASTRIKE_VERSION_MAJOR}.${VEGASTRIKE_VERSION_MINOR}.${VEGASTRIKE_VERSION_PATCH}")
```
Comment (BenjamenMeyer) [inline engine/CMakeLists.txt:63]: FYI - I used the `_STR` to denote that the values were string values where the other data could be used as integer values in other parts of the pipeline.

**Comment by BenjamenMeyer** [comment]
Don't forget about https://github.com/vegastrike/Vega-Strike-Engine-Source/blob/master/engine/src/version.h.in which also incorporates this data.

File: engine/CMakeLists.txt [hunk 2]
```diff
@@ -33,24 +34,33 @@
...
+IF (DEFINED ENV{TAG_NAME} AND NOT "$ENV{TAG_NAME}" STREQUAL "")
+    STRING(REGEX REPLACE "^v([0-9]+)\\..*" "\\1" VEGASTRIKE_VERSION_MAJOR "$ENV{TAG_NAME}")
+    STRING(REGEX REPLACE "^v[0-9]+\\.([0-9]+).*" "\\1" VEGASTRIKE_VERSION_MINOR "$ENV{TAG_NAME}")
+    STRING(REGEX REPLACE "^v[0-9]+\\.[0-9]+\\.([0-9]+).*" "\\1" VEGASTRIKE_VERSION_PATCH "$ENV{TAG_NAME}")
+    STRING(REGEX REPLACE "^v[0-9]+\\.[0-9]+\\.[0-9]+(.*)" "\\1" VEGASTRIKE_VERSION_TWEAK "$ENV{TAG_NAME}")
+ELSEIF (DEFINED ENV{GITHUB_SHA} AND NOT "$ENV{GITHUB_SHA}" STREQUAL "")
+    SET(VEGASTRIKE_VERSION_MAJOR "0")
+    SET(VEGASTRIKE_VERSION_MINOR "8")
+    SET(VEGASTRIKE_VERSION_PATCH "0")
+    SET(VEGASTRIKE_VERSION_TWEAK "-$ENV{GITHUB_SHA}")
```
[the remark this answers is not in the document]
Comment (stephengtuggy) [reply engine/CMakeLists.txt]: @BenjamenMeyer How's this?

File: engine/CMakeLists.txt (hunk 1 above)
Comment (stephengtuggy) [reply engine/CMakeLists.txt:56]: OK, changed it to `VEGASTRIKE_VERSION_LONG_STR` and `VEGASTRIKE_VERSION_SHORT_STR`. Does that work?

**Review by BenjamenMeyer — APPROVED** [review]
LGTM

**Pull request merged**